Source/Core/DiscIO/VolumeVerifier.h

```diff
@@ -36,6 +36,72 @@ namespace DiscIO
 {
 class FileInfo;
 
+template <typename T>
+struct Hashes
+{
+  T crc32;
+  T md5;
+  T sha1;
+};
+
+class RedumpVerifier final
+{
+public:
+  enum class Status
+  {
+    Unknown,
+    GoodDump,
+    BadDump,
+    Error,
+  };
+
+  struct Result
+  {
+    Status status = Status::Unknown;
+    std::string message;
+  };
+
+  void Start(const Volume& volume);
+  Result Finish(const Hashes<std::vector<u8>>& hashes);
+
+private:
+  enum class DownloadStatus
+  {
+    NotAttempted,
+    Success,
+    Fail,
+    FailButOldCacheAvailable,
+    SystemNotAvailable,
+  };
+
+  struct DownloadState
+  {
+    std::mutex mutex;
+    DownloadStatus status = DownloadStatus::NotAttempted;
+  };
+
+  struct PotentialMatch
+  {
+    u64 size;
+    Hashes<std::vector<u8>> hashes;
+  };
+
+  static DownloadStatus DownloadDatfile(const std::string& platform, DownloadStatus old_status);
+  static std::vector<u8> ReadDatfile(const std::string& platform);
+  std::vector<PotentialMatch> ScanDatfile(const std::vector<u8>& data);
+
+  std::string m_game_id;
+  u16 m_revision;
+  u8 m_disc_number;
+  u64 m_size;
+
+  std::future<std::vector<PotentialMatch>> m_future;
+  Result m_result;
+
+  static DownloadState m_gc_download_state;
```

Comment:
These can be `static inline` to avoid the need to redefine them in the cpp file.

Reply:
I'm getting build errors on non-Windows now. Can this be resolved without removing the `inline`?

Reply:
I've reverted the `inline` change for now.